engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java

```diff
@@ -239,8 +243,17 @@ void discardLogLines( TransExecutorData transExecutorData ) {
     // TODO: make this optional/user-defined later
     Trans executorTrans = transExecutorData.getExecutorTrans();
     if ( executorTrans != null ) {
-      KettleLogStore.discardLines( executorTrans.getLogChannelId(), false );
-      LoggingRegistry.getInstance().removeIncludingChildren( executorTrans.getLogChannelId() );
+      String logChannelId = executorTrans.getLogChannelId();
+      Timer discardLinesTimer = new Timer( "Trans executor discard lines timer" );
+      TimerTask discardLinesTask = new TimerTask() {
+        @Override public void run() {
+          KettleLogStore.discardLines( logChannelId, false );
+          LoggingRegistry.getInstance().removeIncludingChildren( logChannelId );
+        }
```

Comment:
Nothing cancels the timer - so it will just keep running for an arbitrary amount of time until it's GCd. As I understand it, it can even keep the application from terminating. Seems to me you need a mechanism for invoking the cancel method of the timer.